std/path.d

```diff
@@ -3883,11 +3886,16 @@ string expandTilde(string inputPath) nothrow
                         break;
                     }
 
-                    if (errno != ERANGE)
+                    if (errno != ERANGE &&
+                        // On FreeBSD and OSX, errno can be left at 0 instead of set to ERANGE
+                        errno != 0)
```

Comment:
Shouldn't this check come inside the if clause above? I mean, I didn't think `errno` had any specified meaning if your return from the syscall was 0. It's always "if this returns nonzero, check `errno`".